lib/src/clojure_lsp/queries.clj

```diff
@@ -24,24 +23,22 @@
     nil
     coll))
 
-(defn filter-project-analysis-xf [db]
-  (let [source-paths (settings/get db [:source-paths])]
-    (remove #(shared/external-filename? (first %) source-paths))))
+(defn filter-project-analysis-xf []
+  (remove (comp :external? first second)))
```

Comment:
This took me a few minutes to figure out. Could probably use a comment explaining that it's using the first element as a proxy for whether the file is external.

Reply:
Also, if it works, `(comp :external? first val)` might be slightly clearer.